examples/jsm/loaders/GLTFLoader.js

```diff
@@ -3659,6 +3751,235 @@ class GLTFParser {
 
 	}
 
+	tryResolveNodeId( path, type ) {
+
+		let name = '';
+		if ( type === 'node' ) {
+
+			name = path.substring( '/nodes/'.length );
+
+		} else if ( type === 'material' ) {
+
+			name = path.substring( '/materials/'.length );
+
+		} else if ( type === 'light' ) {
+
+			name = path.substring( '/extensions/KHR_lights_punctual/lights/'.length );
+
+		} else if ( type === 'camera' ) {
+
+			name = path.substring( '/cameras/'.length );
+
+		}
+
+		name = name.substring( 0, name.indexOf( '/' ) );
+		const index = Number.parseInt( name );
+		return index;
+
+	}
+
+	getAnimationPointerDependency( target ) {
+
+		let targetProperty = undefined;
+
+		// check if this is a extension animation
+		let type = TARGET_TYPE.node;
+		let targetId = undefined;
+
+		const useExtension = target.extensions && target.extensions[ EXTENSIONS.KHR_ANIMATION_POINTER ];
+		if ( useExtension ) {
+
+			const ext = target.extensions[ EXTENSIONS.KHR_ANIMATION_POINTER ];
+			let path = ext.path;
+			if ( animationPointerDebug )
+				console.log( 'Original path: ' + path );
+
+			if ( ! path ) {
+
+				console.warn( 'Invalid path', ext, target );
+				return;
+
+			}
+
+			if ( path.startsWith( '/materials/' ) )
+				type = TARGET_TYPE.material;
+			else if ( path.startsWith( '/extensions/KHR_lights_punctual/lights/' ) )
+				type = TARGET_TYPE.light;
+			else if ( path.startsWith( '/cameras/' ) )
+				type = TARGET_TYPE.camera;
+
+			targetId = this.tryResolveNodeId( path, type );
+			if ( targetId === null || isNaN( targetId ) ) {
+
+				console.warn( 'Failed resolving animation node id: ' + targetId, path );
+				return;
+
+			} else {
+
+				if ( animationPointerDebug ) console.log( 'Resolved node ID for ' + type, targetId );
+
+			}
+
+			// TODO could be parsed better
+			switch ( type ) {
+
+				case TARGET_TYPE.material:
+					const pathIndex = ( '/materials/' + targetId.toString() + '/' ).length;
+					const pathStart = path.substring( 0, pathIndex );
+					targetProperty = path.substring( pathIndex );
+
+					// TODO implement all properties and/or find a better way to have a good mapping here
```

Comment:
I suspect "implement all properties" may need to be a non-goal for us, especially if/when animated extension properties come into the mix. We can try to support whichever properties seem helpful to three.js users and not overly-complex to include.

No need to change these in the draft PR now, but a few things to flag:

- metallicFactor seems unlikely to be animated much?
- alphaCutoff animation can trigger shader recompilation 
- we have very limited support for UV transforms, all textures except aoMap must share the same transform. I wonder if we should avoid animating those properties until that is improved.